Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -178,6 +179,15 @@ export class azureclitask {
         }
     }
 
+    private static async setAzureCloudBasedOnServiceEndpoint(): Promise<void> {
+        var connectedService: string = tl.getInput("connectedServiceNameARM", true);
+        const endpoint: AzureEndpoint = await new AzureRMEndpoint(connectedService).getEndpoint();
```

Comment:
You can also do "tl.getEndpointDataParameter(connectedService, 'environment', true)" instead of taking dependency on azure-arm-rest

Reply:
Done.